models/taskRequests.js

```diff
@@ -189,85 +189,95 @@ const fetchTaskRequestById = async (taskRequestId) => {
-    if (isCreationRequestApproved) {
-      return { isCreationRequestApproved };
-    }
-    const userRequest = {
-      userId: data.userId,
-      proposedDeadline: data.proposedDeadline,
-      proposedStartDate: data.proposedStartDate,
-      description: data.description,
-      status: TASK_REQUEST_STATUS.PENDING,
-    };
-    if (!userRequest.description) delete userRequest.description;
-    if (taskRequestData) {
-      // TODO : remove after the migration of old data https://github.com/Real-Dev-Squad/website-backend/issues/1613
-      const currentRequestors = taskRequestData.requestors;
-      let alreadyRequesting = currentRequestors.some((requestor) => requestor === data.userId);
-      // End of old logic
-      const currentRequestingUsers = taskRequestData.users;
-      alreadyRequesting = currentRequestingUsers.some((requestor) => requestor.userId === data.userId);
-      if (alreadyRequesting) {
-        return { alreadyRequesting };
+    return await firestore.runTransaction(async (transaction) => {
+      const queryFieldPath = data.requestType === TASK_REQUEST_TYPE.CREATION ? "externalIssueUrl" : "taskId";
+      const queryValue = data.requestType === TASK_REQUEST_TYPE.CREATION ? data.externalIssueUrl : data.taskId;
+      const statusQueryValue =
+        data.requestType === TASK_REQUEST_TYPE.CREATION
+          ? [TASK_REQUEST_STATUS.PENDING, TASK_REQUEST_STATUS.APPROVED]
+          : [TASK_REQUEST_STATUS.PENDING];
+
+      const taskRequestsDocRef = taskRequestsCollection
+        .where(queryFieldPath, "==", queryValue)
+        .where("status", "in", statusQueryValue);
+      const taskRequestsSnapshot = await transaction.get(taskRequestsDocRef);
+
+      const [taskRequestRef] = taskRequestsSnapshot.docs;
+      const taskRequestData = taskRequestRef?.data();
+      const isCreationRequestApproved =
+        taskRequestData &&
+        taskRequestData.requestType === TASK_REQUEST_TYPE.CREATION &&
+        taskRequestData.status === TASK_REQUEST_STATUS.APPROVED;
+      if (isCreationRequestApproved) {
+        return { isCreationRequestApproved };
       }
-      // TODO : remove after the migration of old data https://github.com/Real-Dev-Squad/website-backend/issues/1613
-      const updatedRequestors = [...currentRequestors, data.userId];
-      // End of old logic
-      const updatedUsers = [...currentRequestingUsers, userRequest];
-      const updatedTaskRequest = {
-        requestors: updatedRequestors,
-        users: updatedUsers,
+      const userRequest = {
+        userId: data.userId,
+        proposedDeadline: data.proposedDeadline,
+        proposedStartDate: data.proposedStartDate,
+        description: data.description,
+        status: TASK_REQUEST_STATUS.PENDING,
+      };
+      if (!userRequest.description) delete userRequest.description;
+      if (taskRequestData) {
+        // TODO : remove after the migration of old data https://github.com/Real-Dev-Squad/website-backend/issues/1613
```

Comment:
Nit : since the issue is closed `https://github.com/Real-Dev-Squad/website-backend/issues/1613` the code dealing with migration of old data can be removed.

Reply:
Will remove these in separate PR while removing dev flag.